### PR TITLE
Add lint workflow to run pre-commit on all files

### PR DESCRIPTION
**Pull request by oliverholworthy**
### Goals :soccer:

Make local linting with pre-commit consistent with lint checks run in CI.

### Implementation Details :construction:

Run pre-commit on all files in a lint workflow

### Testing Details :mag:

check lint workflow

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ repos:
         hooks:
           - id: mypy
             language_version: python3
-            args: [--no-strict-optional, --ignore-missing-imports, --show-traceback, --install-types, --non-interactive]
+            args: [--non-interactive, --install-types]
```
Comment (oliverholworthy) [inline .pre-commit-config.yaml:20]: the rest of these args have moved to `mypy.ini`

File: merlin_standard_lib/schema/schema.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
     # polyfill cached_property for python <= 3.7 (using lru_cache which was introduced in python3.2)
     from functools import lru_cache
 
-    cached_property = lambda func: property(lru_cache()(func))  # noqa
+    cached_property = lambda func: property(lru_cache()(func))  # type: ignore  # noqa
```
Comment (oliverholworthy) [inline merlin_standard_lib/schema/schema.py:30]: mypy throws errors about assigning to types here:

```
merlin_standard_lib/schema/schema.py:30: error: Cannot assign to a type
merlin_standard_lib/schema/schema.py:30: error: Incompatible types in assignment (expression has type "Type[property]", variable has type "Type[cached_property[Any]]")
merlin_standard_lib/schema/schema.py:30: error: Incompatible return value type (got "property", expected "cached_property[_T]")
```

File: mypy.ini [hunk 3]
```diff
@@ -1,7 +1,10 @@
 [mypy]
-python_version = 3.7
+python_version = 3.8
```
Comment (oliverholworthy) [inline mypy.ini:2]: Updating python version to 3.8 now that we have a min version of 3.8

File: requirements/dev.txt [hunk 4]
```diff
@@ -1,8 +1,2 @@
 check-manifest
-black==20.8b1
```
Comment (oliverholworthy) [inline requirements/dev.txt:2]: These are now configured  only in `.pre-commit-config.yaml`

File: setup.cfg [hunk 5]
```diff
@@ -28,7 +28,7 @@ tag_prefix = v
 parentdir_prefix = transformers4rec-
 
 [codespell]
-skip = .*pb2.py,./.git,./.github,./.mypy_cache,./dist,./build,./docs/build,.*egg-info.*,versioneer.py,*.csv,*.parquet
+skip = .*pb2.py,.git/*,.github/*,./.mypy_cache,./dist,./build,./docs/build,.*egg-info.*,versioneer.py,*.csv,*.parquet
```
Comment (oliverholworthy) [inline setup.cfg:31]: `./.github` wasn't working to exclude this directory. Changed to `.github/*` and it works to ignore

File: tests/data/test_preprocessing.py [hunk 6]
```diff
@@ -21,7 +18,7 @@ def test_remove_consecutive_interactions():
     schema += Schema([ColumnSchema.create_continuous("timestamp", tags=[Tag.SESSION])])
 
     interactions_df = generate_item_interactions(500, schema)
-    filtered_df = remove_consecutive_interactions(interactions_df.copy())
+    filtered_df = preprocessing.remove_consecutive_interactions(interactions_df.copy())
```
Comment (oliverholworthy) [inline tests/data/test_preprocessing.py:21]: my was throwing an error about the module not having the attribute with the function import version (not sure I fully understand why). But this seems to work without error:

```
tests/data/test_preprocessing.py:5: error: Module "transformers4rec.data.preprocessing" has no attribute "add_item_first_seen_col_to_df"  [attr-defined]
tests/data/test_preprocessing.py:5: error: Module "transformers4rec.data.preprocessing" has no attribute "remove_consecutive_interactions"  [attr-defined]
```

**Review by karlhigley — APPROVED** [review]

**Pull request merged**